homeassistant/components/loqed/lock.py

```diff
@@ -0,0 +1,269 @@
+    )
+    _LOGGER.debug(
+        "Inititated loqed-lock entity with id: %s and host: %s", lock.id, host
+    )
+    if not lock:
+        # No locks found; abort setup routine.
+        _LOGGER.info(
+            "We cannot connect to the loqed lock, \
+                please try to reinstall the integation"
+        )
+        return
+    async_add_entities([LoqedLock(lock, data["internal_url"], "http://" + host)])
+
+
+def get_random_string(length):
+    """Create a rondom ascii string."""
+    letters = string.ascii_lowercase
+    result_str = "".join(random.choice(letters) for i in range(length))
+    return result_str
+
+
+class LoqedLock(LockEntity):
+    """Representation of a loqed lock."""
+
+    def __init__(self, lock: loqed.Lock, internal_url, lock_url) -> None:
+        """Initialize the lock."""
+        self.lock_url = lock_url
+        self._lock = lock
+        self._internal_url = internal_url
+        self._webhook = ""
+        self._attr_unique_id = self._lock.id
+        self._attr_name = self._lock.name
+        self._attr_supported_features = SUPPORT_OPEN
+        # self._attr_supported_features = LockEntityFeature.OPEN
+        self.update_task = None
+
+    async def async_added_to_hass(self) -> None:
+        """Entity created."""
+        await super().async_added_to_hass()
+        await self.check_webhook()
+
+    @property
+    def changed_by(self):
+        """Return true if lock is locking."""
+        return "KeyID " + str(self._lock.last_key_id)
+
+    @property
+    def bolt_state(self):
+        """Return true if lock is locking."""
+        return self._lock.bolt_state
+
+    @property
+    def is_locking(self):
+        """Return true if lock is locking."""
+        return LOCK_STATES[self.bolt_state] == STATE_LOCKING
+
+    @property
+    def is_unlocking(self):
+        """Return true if lock is unlocking."""
+        return LOCK_STATES[self.bolt_state] == STATE_UNLOCKING
+
+    @property
+    def is_jammed(self):
+        """Return true if lock is jammed."""
+        return LOCK_STATES[self.bolt_state] == STATE_JAMMED
+
+    @property
+    def is_locked(self):
+        """Return true if lock is locked."""
+        return LOCK_STATES[self.bolt_state] == STATE_LOCKED
+
+    @property
+    def battery(self):
+        """Return true if lock is locked."""
+        return self._lock.battery_percentage
+
+    @property
+    def extra_state_attributes(self):
+        """Extra state attribtues."""
+        state_attr = {
+            "id": self._lock.id,
+            "bolt_state": self.bolt_state,
+            "lock_url": self.lock_url,
+            "webhook_url": self._webhook,
+            ATTR_BATTERY_LEVEL: self._lock.battery_percentage,
+            "battery_type": self._lock.battery_type,
+            "battery_voltage": self._lock.battery_voltage,
+            "wifi_strength": self._lock.wifi_strength,
+            "ble_strength": self._lock.ble_strength,
+            "last_event": self._lock.last_event,
```

Comment:
Lots of these attributes should become their own entities.
Attributes are rather expensive and inefficient. They are also hard to use for an end user.

Also, consider what is needed. For example, is `id` of a lock useful for the end user?